### DISCUSSION TITLE
How to debounce a model change?

### DISCUSSION THREAD

**Discussion by antonmedv** [charmbracelet/bubbletea#601] [Q&A] [answered]
I’d like to debounce a func call after a key stroke. I found a tea.Tick, but I need update the model instantly. And clear some model field after one sec. 

Is it safe to modify a model from a goruthine?

**Comment by meowgorithm** [comment, marked as answer]
There’s actually a simple pattern for this which is probably best explained with an example:

https://github.com/charmbracelet/bubbletea/blob/master/examples/debounce/main.go

But do let us know if you have more questions, too.

**Reply by antonmedv** [reply]
```go
		// This will delay full rerender, what to do if I need to do full rerender?      
		// eg return m, nil?
		// And have a debounce as well.           
		return m, tea.Tick(debounceDuration, func(_ time.Time) tea.Msg {
			// ...and include a copy of that tag value in the message.
			return exitMsg(m.tag)
		})
```

**Reply by meowgorithm** [reply]
Since `Update` is returning here `View` will still be called immediately after returning and any necessary redraws will be handled by the renderer.

**Reply by antonmedv** [reply]
Thanks for explanation!
